hack/go-tools-build.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [ -z "${1}" ]; then
+  echo "must provide module as first parameter"
+  exit 1
+fi
+
+if [ -z "${2}" ]; then
+  echo "must provide binary name as second parameter"
+  exit 1
+fi
+
+if [ -z "${3}" ]; then
+  echo "must provide version as third parameter"
+  exit 1
+fi
+
+if [ -z "${GOBIN}" ]; then
+  echo "GOBIN is not set. Must set GOBIN to install the bin in a specified directory."
+  exit 1
+fi
+
+rm -f "${GOBIN}/${2}"* || true
+
+ORIGINAL_WORKDIR="$(pwd)"
+TMP_MODULE_DIR="${2}.tmp"
+
+# Create TMP_MODULE_DIR to create a go module for building the binary.
+rm -r "${TMP_MODULE_DIR}" || true
+mkdir -p "${TMP_MODULE_DIR}"
+cd "${TMP_MODULE_DIR}"
+
+# Initialize a go module and place a tools.go file for building the binary.
+go mod init "tools"
+# Set require for "sigs.k8s.io/cluster-api" to let go resolve the tools version via the CAPI version.
+go mod edit -replace "sigs.k8s.io/controller-runtime/tools/setup-envtest=github.com/sbueringer/controller-runtime/tools/setup-envtest@v0.0.0-20240506183901-d950bb92291d"
```

Comment:
Using the version from my PR until it is merged. Going to change this to latest commit on CR main before this PR here is merged

@chrischdi Thx for the script in CAPV btw that I copied almost all of this from :)